lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -144,7 +144,96 @@ TODO
 
 ## Using exceptions for control-flow
 
-TODO
+#### Problem
+
+This anti-pattern refers to code that forces developers to handle exceptions for control flow. Exception handling itself does not represent an anti-pattern, but this should not be the only alternative available to developers to handle an error in client code. When developers have no freedom to decide if an error is exceptional or not, this is considered an anti-pattern.
+
+#### Example
+
+An example of this anti-pattern, as shown below, is when a library forces its users to use `try/1` statements to rescue raised exceptions and handle different cases. Such a library library does not allow developers to decide if an error is exceptional or not in their applications.
+
+```elixir
+defmodule MyModule do
+  def janky_function(value) do
+    if is_integer(value) do
+      "It's an integer"
+    else
+      raise "expected integer, got: #{inspect(value)}"
+    end
+  end
+end
+```
+
+```elixir
+defmodule Client do
+  # Client forced to use exceptions for control flow.
+  def print_janky_function(arg) do
+    try do
+      result = MyModule.janky_function(arg)
+      "All good! #{result}."
+    rescue
+      exception in RuntimeError ->
+        "Uh oh! #{exception.message}."
+    end
+  end
+end
+```
+
+```elixir
+iex> Client.print_janky_function(1)
+"All good! It's an integer."
+iex> Client.foo("Lucas")
+"Uh oh! expected integer, got: \"Lucas\""
+```
+
+#### Refactoring
+
+Library authors should guarantee that users are not required to use exceptions for control flow in their applications. As shown below, this can be done by refactoring `MyModule`, providing two versions of the function that forces clients to use exceptions for control flow: 
```

Comment:
```suggestion
Library authors should guarantee that users are not required to use exceptions for control flow in their applications. As shown below, this can be done by refactoring `MyModule`, providing two versions of the function that forces clients to use exceptions for control flow:
```